src/kmergraph.cpp

```diff
@@ -238,20 +218,271 @@ void KmerGraph::discover_k() {
     }
 }
 
-void KmerGraph::set_exp_depth_covg(const uint32_t edp) {
+std::ostream &operator<<(std::ostream &out, KmerGraph const &data) {
+    for (const auto &c: data.nodes) {
+        out << *(c) << std::endl;
+    }
+    return out;
+}
+
+//save the KmerGraph as gfa
+void KmerGraph::save(const std::string &filepath, const std::shared_ptr<LocalPRG> localprg) {
+    uint32_t sample_id = 0;
+
+    std::ofstream handle;
+    handle.open(filepath);
+    if (handle.is_open()) {
+        handle << "H\tVN:Z:1.0\tbn:Z:--linear --singlearr" << std::endl;
+        for (const auto &c : nodes) {
+            handle << "S\t" << c->id << "\t";
+
+            if (localprg != nullptr) {
+                handle << localprg->string_along_path(c->path);
+            } else {
+                handle << c->path;
+            }
+
+            //TODO: leave as coverage 0 or change this?
+            handle << "\tFC:i:" << 0 << "\t" << "\tRC:i:" << 0 << std::endl;//"\t" << (unsigned)nodes[i].second->num_AT << endl;
+
+            for (uint32_t j = 0; j < c->outNodes.size(); ++j) {
+                handle << "L\t" << c->id << "\t+\t" << c->outNodes[j].lock()->id << "\t+\t0M" << std::endl;
+            }
+        }
+        handle.close();
+    } else {
+        std::cerr << "Unable to open kmergraph file " << filepath << std::endl;
+        std::exit(EXIT_FAILURE);
+    }
+}
+
+void KmerGraph::load(const std::string &filepath) {
+    clear();
+    uint32_t sample_id = 0;
+
+    std::string line;
+    std::vector<std::string> split_line;
+    std::stringstream ss;
+    uint32_t id = 0, covg, from, to;
+    prg::Path p;
+    uint32_t num_nodes = 0;
+
+    std::ifstream myfile(filepath);
```

Comment:
Suggest rename to something like `graph_file`

Reply:
Done in https://github.com/leoisl/pandora/commit/56fe941503e7c83dd5b316b275e04c1104e4b06c